expo/expo
# [home][chore] migrate to react navigation v5

# Why

- fix https://github.com/expo/expo/issues/9344



# How

- Remove old react-navigation
- Migrate to new theme system
- Fix all remaining style bugs on Android
- Migrate more to TypeScript
- Use new theme provider for dev menu



# Test Plan

- Manually tested on physical devices and iOS simulator
  - Ensure the NavigationEvents shim continues to work
  - Ensure the dev menu respects the style defined in home
  - Ensure status bar respects color scheme
